src/codeu/chat/server/Server.java

```diff
@@ -169,6 +193,10 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
 
         Serializers.INTEGER.write(out, NetworkCode.GET_MESSAGES_BY_ID_RESPONSE);
         Serializers.collection(Message.SERIALIZER).write(out, messages);
+      if (type == NetworkCode.SERVER_INFO_REQUEST) {
```

Comment:
See my above comment - you only need to make that change once, so you can probably delete this section.